#goal constants
uint8 STOP=0
uint8 READY=1
uint8 EXPAND=2
uint8 CONTRACT=3
uint8 CLAMP=4
uint8 UNCLAMP=5
uint8 EXPAND_AND_UNCLAMP=6
uint8 CLAMP_AND_CONTRACT=7
uint8 EXPAND_AND_PLACE=8
uint8 EXPAND_AND_PLACE_AND_CONTRACT=9

#goal fields
uint8 command

---
#result constants
uint8 OK=0
uint8 ABORTED=1
uint8 ERR_UNEXPECTED_ARG=2
uint8 ERR_CANNOT_PROCESS=3

#result fields
uint8 result

---
#feedback constants
uint8 STOPPING=0
uint8 PREV_ACTION_ABORTED=1
uint8 EXPANDED_AT_EXPAND_AND_UNCLAMP=2
uint8 CLAMPED_AT_CLAMP_AND_CONTRACT=3
uint8 EXPANDED_AT_EXPAND_AND_PLACE=4
uint8 EXPANDED_AT_EXPAND_AND_PLACE_AND_CONTRACT=5
uint8 PLACE_AT_EXPAND_AND_PLACE_AND_CONTRACT=6

#feedback fields
uint8 feedback
